editor/themes/theme.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "common/types.h"
+#include "../../rootex/vendor/JSON/json.hpp"
+#include <fstream>
+
+struct ThemeDefinition
+{
+	String m_Name;
+	String m_filePath;
+	nlohmann::json m_ThemeData;
+
+	void apply() const
+	{
+		ImGui::StyleColorsDark(); // Base reset
+
+		ImGuiStyle& style = ImGui::GetStyle();
+		ImVec4* colors = style.Colors;
+
+		// Only apply if present
+		if (!m_ThemeData.contains("style") && !m_ThemeData.contains("colors"))
+		{
+			return;
+		}
+
+		// --- Lambda utilities ---
+		auto setVec2 = [](const nlohmann::json& j, const char* key, ImVec2& ref)
+		{
+			if (j.contains(key) && j[key].is_array() && j[key].size() == 2)
+				ref = ImVec2(j[key][0], j[key][1]);
+		};
+
+		auto setFloat = [](const nlohmann::json& j, const char* key, float& ref)
+		{
+			if (j.contains(key))
+				ref = j[key].get<float>();
+		};
+
+		auto setColor = [](const nlohmann::json& j, const char* key, ImVec4& ref)
+		{
+			if (j.contains(key) && j[key].is_array() && j[key].size() == 4)
+				ref = ImVec4(j[key][0], j[key][1], j[key][2], j[key][3]);
```

Comment:
Good to have some alert (not a error but some warning) in case the .json file is corrupted.